### ISSUE TITLE
wrong output when use PyWren inside PyWren

### ISSUE THREAD

**Issue by omerb01** [pywren/pywren-ibm-cloud#181]
I get an unexpected output when I run this code for example:
```python
import pywren_ibm_cloud as pywren

def my_map_function(x):
    def _func(x):
        return x + 1

    pw = pywren.ibm_cf_executor()
    pw.map(_func, range(10))
    return pw.get_result()

pw = pywren.ibm_cf_executor()
pw.map(my_map_function, range(10))
results = pw.get_result()
```
sometimes I get lists inside `results` and sometimes everything is fine.
for example:
![Screen Shot 2019-08-06 at 15 45 29](https://user-images.githubusercontent.com/13593227/62540908-5346d400-b861-11e9-8e03-d481bb172734.png)

**Comment by gilv** [comment]
@JosepSampe can you see it please?

**Comment by gilv** [comment]
@JosepSampe I assume this code suppose to generate final result which is 10 elements, where each element is a list of 10 elements, right?

**Comment by JosepSampe** [comment]
@gilv Yep, It should return 10 lists of 10 elements. I already found the source of the problem, I'm solving it ;)

**Comment by JosepSampe** [comment]
@omerb01 @gilv Fixed here #180

**Comment by JosepSampe** [comment]
Fixed here #180

@omerb01 It will be nice if you add this example into tests ;)
